### PR TITLE
Implementation of Npath complexity for the OpenCover reports

### LINKED ISSUE #1057: Opencover nPathComplexity is hardcoded to 0
When creating an OpenCover report the nPathComplexity attribute is hard coded to 0.

### PR DESCRIPTION
**Pull request by benjaminZale**
Resolves #1057 and #132:

Npath complexity is now calculated for OpenCover Reports

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
Can you confirm you've adapted this https://github.com/OpenCover/opencover/blob/fa166a24fa0c0f00a62ac140b1a48cdc6aacdff1/main/OpenCover.Framework/Persistance/BasePersistance.cs#L419?

**Comment by benjaminZale** [comment]
> 
> 
> Can you confirm you've adapted this https://github.com/OpenCover/opencover/blob/master/main/OpenCover.Framework/Persistance/BasePersistance.cs#L419 ?
> 
> If so can you add comment+link above method? i.e. "Taken from...."

Done

File: src/coverlet.core/CoverageSummary.cs [hunk 1]
```diff
@@ -73,6 +73,39 @@ public CoverageDetails CalculateBranchCoverage(IList<BranchInfo> branches)
...
+            int npath = 1;
+            foreach (var branchPoints in paths.Values)
+            {
+                try
+                {
+                    npath = checked(npath * branchPoints);
+                }
+                catch (OverflowException)
+                {
+                    npath = int.MaxValue;
```
Comment (MarcoRossignoli) [inline src/coverlet.core/CoverageSummary.cs:103]: why not `break` here?(like OpenCover alg)

**Comment by MarcoRossignoli** [comment]
Can you update tests https://github.com/coverlet-coverage/coverlet/blob/master/test/coverlet.core.tests/Reporters/CoberturaReporterTests.cs?

**Comment by MarcoRossignoli** [comment]
@benjaminZale any news on this?

**Review by MarcoRossignoli — APPROVED** [review]
LGTM Thanks!

**Pull request merged**